daq/host.py

```diff
@@ -629,10 +615,15 @@ def _device_aux_path(self):
             os.makedirs(path)
         return path
 
-    def _docker_test(self, test_name):
-        self.test_host = docker_test.DockerTest(self.runner, self.target_port,
-                                                self.devdir, test_name)
-        self.logger.debug('test_host start %s/%s', test_name, self._host_name())
+    def _new_test(self, test_name):
+        clazz = ipaddr_test.IpAddrTest if test_name == 'ipaddr' else docker_test.DockerTest
```

Comment:
So IpAddrTest is a single port toggle test inside the ipaddr module whereas the DockerTest is a module, since Host class should be host of modules, something is not aligned.  I think there should also be the test abstraction and module abstraction(other than dockertest) in place first before the current port_toggle test class. 
Ip addr test and DockerTest are also both named test which is why we should also spend some time clearly distinguish test and module in code when we have time for more refactoring.

Reply:
IpAddrTest and DockerTest are both "modules" in the sense they are a high-level entity that's managed by ConnectedHost. Both IpAddrTest and DockerTest, in turn, encapsulate individual line-item tests.  There is an abstraction in place, but maybe you mean that it should be an _explicit_ abstraction (e.g. with a named base-class or similar?)